File: app/src/main/java/ru/evotor/framework/core/action/event/inventory/ProductEvent.java
package ru.evotor.framework.core.action.event.inventory;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public abstract class ProductEvent {
    private static final String KEY_PRODUCT_UUID = "productUuid";

    @Nullable
    private final String productUuid;

    public ProductEvent(Bundle extras) {
        this(
                extras.getString(KEY_PRODUCT_UUID)
        );
    }

    public ProductEvent(@Nullable String productUuid) {
        this.productUuid = productUuid;
    }

    @NonNull
    public Bundle toBundle() {
        Bundle result = new Bundle();
        result.putString(KEY_PRODUCT_UUID, productUuid);
        return result;
    }

    @Nullable
    public String getProductUuid() {
        return productUuid;
    }
}

File: src/main/java/ru/evotor/framework/inventory/event/ProductEvent.kt
package ru.evotor.framework.inventory.event

import android.os.Bundle

import ru.evotor.IBundlable

abstract class ProductEvent internal constructor(val productUuid: String?) : IBundlable {

    override fun toBundle(): Bundle {
        val result = Bundle()
        result.putString(KEY_PRODUCT_UUID, productUuid)
        return result
    }

    companion object {

        private const val KEY_PRODUCT_UUID = "productUuid"

        internal fun getProductUuid(bundle: Bundle): String? = bundle.getString(KEY_PRODUCT_UUID)

    }

}
